### ISSUE TITLE
ESC key stops working with Drag Ruler installed

### ISSUE THREAD

**Issue by simulacrum-1** [manuelVo/foundryvtt-drag-ruler#113]
If both modules are active the Esc Key does not function.
DFreds Droppables 1.2.10
Drag Ruler 1.8.1

**Comment by simulacrum-1** [comment]
Though it could be just drag ruler. As with drag ruler turned off the Esc key works. On Chrome, Forge on a mac.

**Comment by manuelVo** [comment]
What's the expected behavior when pressing the Esc key in DFreds Droppables?

**Comment by simulacrum-1** [comment]
I spoke with DFreds and the module does nothing with the escape key, or any key at all. So I think it must just be Drag Ruler Causing it.

**Comment by manuelVo** [comment]
What's the thing about the Escape key that's breaking then?

**Comment by manuelVo** [comment]
@gabrieljtaylor Can you elaborate on what functionality you'd expect the ESC Key to perform, that stops working with Drag Ruler enabled? Closing windows and opening the ESC menu (both core functionality) work fine for me on my test rig.

**Comment by gabrieljtaylor** [comment]
@manuelVo - ALL escape key functionality is not working for me, but only when Drag Ruler is enabled. When I disable the module, the ESC key functions as normal, but when Drag Ruler is enabled, clicking the button does absolutely nothing - no windows are closed, the ESC menu doesn't open, etc. Literally nothing occurs when I press the ESC key while Drag Ruler is enabled.

Are there any particular logs or information I can send you that may help you trace the issue?

**Comment by manuelVo** [comment]
First thing to look at is the JS console (opens with F12). Are there any Errors (red) or warnings (yellow) in there - either after loading or after you've pressed ESC? If so, please screenshot them and post them here.

I'm fairly certain that this is a conflict with another module, since it works well in my isolated test world. Please use the "Find the culprit" module and mark "Drag Ruler" and "socketlib" as working modules when starting the process. Check which module is reported by "Find the Culprit" as broken. That module is very likely the conflicting module.

**Comment by gabrieljtaylor** [comment]
It appears that the conflict may be related to libWrapper or the usage thereof - the error message is being generated by libWrapper and listing Drag Ruler, libWrapper itself, and an unnamable module as those involved. Both Drag Ruler and the unnamable module work when only they and libWrapper are enabled, but they don't work together and it isn't clear why.

The unnamable module only recently implemented libWrapper support, so I'm now guessing the issue may have something to do with their implementation, and thus am running down a support request on that side until more information is available. I'll also put the text of the error message below in case it is useful to you:

```
libWrapper-wrapper.js:672 

       Uncaught TypeError: Cannot read properties of null (reading 'draggedEntity')
[Detected 3 packages: drag-ruler, unnamable-module, lib-wrapper]
    at Object.cancelDrag [as onDown] (keybindings.js:84)
    at Function._executeKeybind (foundry.js:1595)
    at KeyboardManager._processKeyboardContext (foundry.js:1634)
    at KeyboardManager._handleKeyboardEvent (foundry.js:1672)
    at Z.🎁call_wrapped [as call_wrapped] (libWrapper-wrapper.js:510)
    at KeyboardManager.🎁game.keyboard._handleKeyboardEvent#unnamable-module (PatcherGameKeyboard.js:26)
    at Z.🎁call_wrapper [as call_wrapper] (libWrapper-wrapper.js:619)
    at KeyboardManager.🎁game.keyboard._handleKeyboardEvent#0 (libWrapper-wrapper.js:188)
    at foundry.js:1335
```

**Comment by manuelVo** [comment]
Definitely a conflict with another module. Could you do the Find the Culprit-Thing I sugfested in my previous message to determine which module conflicts?

**Comment by SimonasBanys** [comment]
just to update on this, I have the same issue but only when no scene is active, if there is a scene active, ESC key works as expected

**Comment by gabrieljtaylor** [comment]
On further testing, my situation is the same as that of @SimonasBanys - the issue only occurs when there are no scenes present or activated. Once a scene is created, the issue disappears, but in a new, sceneless world, it happens consistently. I have also been able to reproduce the error message SimonasBanys posted.

I used FTC twice, once with no modules flagged and another with socketlib and libWrapper flagged (to test with and without dependencies); in both instances, FTC reports Drag Ruler as the culprit.

![image](https://user-images.githubusercontent.com/8386413/148552413-cdf63840-007e-4324-bdae-4ed2d1c18471.png)

What other info can we provide for you?
